### ISSUE TITLE
RuntimeException "Could not move right because the last node at this level has already been reached"

### ISSUE THREAD

**Issue by The-Math-Guy** [cternes/openkeepass#45]
Hello

I'm trying to use Openkeepass to manage my KeePass 2.x databases but I'm getting a 

> java.lang.RuntimeException with message "Could not move right because the last node at this level has already been reached".

Here is the stack trace: [Exception stacktrace.txt](https://github.com/cternes/openkeepass/files/288569/Exception.stacktrace.txt)

My Main class contains only the main method with the instruction 
`KeePassFile db = KeePassDatabase.getInstance(path+"Database.kdbx").openDatabase("MasterPassword");`

I've obfuscated username/passwords/notes/..., changed master password to "MasterPassword" and enclose here the zipped db: [Database.zip](https://github.com/cternes/openkeepass/files/288609/Database.zip)
It's protected only with master password

I've tried to make a quick debug.
I may be wrong but it looks like the issue occurs at the root node level (I get a null parent and a current node name "Database" which is only the root). 

Thanks and regards,
Mika

**Comment by cternes** [comment]
@The-Math-Guy Thanks for opening this issue.

I have issues reproducing your problem. Which version of openkeepass are you using? I've added your sample code and database as a separate unit test into the project but the test works fine.

Check it out by yourself `KeepassDatabaseReaderTest:whenGettingEntriesShouldReturnAllEntries`.

Could you try to execute the test and let me know if it works for you or not?

**Comment by The-Math-Guy** [comment]
I downloaded the last one from here.
After opening this issue I've installed a new eclipse version and unfortunately this caused my whole workspace to be completely deleted .
I downloaded openkeepass 0.5.1 from Maven central repository and I am now able to access my KeePass database.
I guess previous openkeepass installation was not clean or else I have no idea what could have been the problem. 

Thank you for your support!
